common/model/addendum.test.js

```diff
@@ -1,15 +1,20 @@
 import { Addendum, AddendumType } from './addendum'
 import { Identity, IdentityType } from './identity'
+import FirestoreMock from '../test_helpers/firestore.mock'
+import DB from '../db/db'
+
+const signer = new Identity(IdentityType.EMAIL, 'John', 'john@onf.dev')
+const contributor1 = new Identity(IdentityType.EMAIL, 'Emma', 'emma@onf.dev')
+const contributor2 = new Identity(IdentityType.EMAIL, 'Gigi', 'gigi@onf.dev')
+const contributor3 = new Identity(IdentityType.GITHUB, 'Alex', 'alex3000')
 
 describe('The Addendum model', () => {
+  const firestoreMock = new FirestoreMock()
   let model
-  let signer
-  let contributor1, contributor2, contributor3
   beforeEach(() => {
-    signer = new Identity(IdentityType.EMAIL, 'John', 'john@onf.dev')
-    contributor1 = new Identity(IdentityType.EMAIL, 'Emma', 'emma@onf.dev')
-    contributor2 = new Identity(IdentityType.EMAIL, 'Gigi', 'gigi@onf.dev')
-    contributor3 = new Identity(IdentityType.GITHUB, 'Alex', 'alex3000')
+    const DBSpy = jest.spyOn(DB, 'connection').mockImplementation(() => firestoreMock)
```

Comment:
At some point we should be switching to using the firestore emulator here, with no need of mocking DB calls and with the added benefits of making sure we use the right types when using the firestore API. Also, we would be implicitly testing permissions as well..

Reply:
I disagree on this. If we want to add functional tests I'm all for it, but unit tests shouldn't depend on an external database.
Also the advantage of using `mocks` is that you can return different values in different tests, doing that with the emulator will require to populate the DB and clean it after every test (doable but more complex).

Anyway we can move this discussion somewhere else as not related to this PR